compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -228,6 +238,9 @@ public boolean isSameType(BType source, BType target) {
     }
 
     private boolean isSameType(BType source, BType target, Set<TypePair> unresolvedTypes) {
+        if (source == null || target == null) {
+            return false;
```

Comment:
When can these become null? I'm not sure if we should add these checks here, we should probably not call this if a type is null.